ale_linters/javascript/flow.vim

```diff
@@ -0,0 +1,66 @@
+" Author: Zach Perrault -- @zperrault
+" Description: FlowType checking for JavaScript files
+
+let g:ale_javascript_flow_executable =
+\   get(g:, 'ale_javascript_flow_executable', 'flow')
+
+let g:ale_javascript_flow_use_global =
+\   get(g:, 'ale_javascript_flow_use_global', 0)
+
+function! ale_linters#javascript#flow#GetExecutable(buffer) abort
+  if g:ale_javascript_flow_use_global
+    return g:ale_javascript_flow_use_global
+  endif
+
+  return ale#util#ResolveLocalPath(
+  \   a:buffer,
+  \   'node_modules/.bin/flow',
+  \   g:ale_javascript_flow_executable
+  \)
+endfunction
+
+function! ale_linters#javascript#flow#GetCommand(buffer) abort
+  return ale_linters#javascript#flow#GetExecutable(a:buffer)
+  \   . ' check-contents --json --from ale'
+endfunction
+
+function! ale_linters#javascript#flow#Handle(buffer, lines)
+  let l:flow_output = json_decode(join(a:lines, ''))
+
+  if l:flow_output
+    let l:output = []
+
+    for l:error in l:flow_output.errors
+      " Each error is broken up into parts
+      let l:text = ''
+      let l:line = 0
+      for l:message in l:error.message
+        " Comments have no line of column information
+        if l:message.line + 0
+          let l:line = l:message.line + 0
+        endif
+        let l:text = l:text . ' ' . l:message.descr
+      endfor
```

Comment:
Each error has multiple messages. Some of those messages have the proper line numbers and others don't 😕 . I was creating a single message per error. Here's an example of the output I'm parsing.

```json
{
  "flowVersion":"0.33.0",
  "errors":[
    {
      "kind":"infer",
      "level":"error",
      "message":[
        {
          "context":"foo.forEach(() => null)",
          "descr":"property `forEach`",
          "type":"Blame",
          "loc":{
            "source":"-",
            "type":"SourceFile",
            "start":{"line":12,"column":5,"offset":96},
            "end":{"line":12,"column":11,"offset":103}
          },
          "path":"-",
          "line":12,
          "endline":12,
          "start":5,
          "end":11
        },
        {
          "context":null,
          "descr":"Property not found in",
          "type":"Comment",
          "path":"",
          "line":0,
          "endline":0,
          "start":1,
          "end":0
        },
        {
          "context":"foo.forEach(() => null)",
          "descr":"String",
          "type":"Blame",
          "loc":{
            "source":"-",
            "type":"SourceFile",
            "start":{"line":12,"column":1,"offset":92},
            "end":{"line":12,"column":3,"offset":95}
          },
          "path":"-",
          "line":12,
          "endline":12,
          "start":1,
          "end":3
        }
      ]
    }
  ],
  "passed":false
}
```